src/reducers/entities/posts.js

```diff
@@ -666,6 +661,10 @@ export function mergePostBlocks(blocks, posts) {
     // Remove any blocks that may have become empty by removing posts
     nextBlocks = removeEmptyPostBlocks(blocks);
 
+    if (!nextBlocks.length) {
```

Comment:
When would merging post blocks ever remove all of the post blocks?

Reply:
If a channel does not have any posts(Experimental feature where join and leave messages don't exist) its good have the state with empty order and posts as it will have a recent flag which is beneficial in various ways. This check here removes that only block.

Sorry i should have commented few things before turning on the label. My bad. i am gonna through again to see if i can clarify any other tricky parts of the changes here

Reply:
When in doubt, adding comments like that to the code is always good :)

And that makes sense

Reply:
Added the comment in the code